### PR TITLE
2.0.0

### PR DESCRIPTION
**Pull request by svenvandescheur**
Replaces aurelia-http-client with axios.

As much API/interface changes as possible are normalized using abstraction. Therefore this should™ mostly be a backwards compatible change. This will, however, be a major version bump.

### CODE REVIEW

File: src/abstract-consumer.js [hunk 1]
```diff
@@ -52,11 +50,12 @@ export class AbstractConsumer {
         /** {string[]} Keys on this.objectClass that should not be passed to the API. */
         this.unserializableFields = ['__consumer__'];
 
+        /** {HttpClient} The Aurelia HttpClient instance to work with. */
+        this.client = new AxiosHTTPClient({baseURL: endpoint, headers: this.defaultHeaders}); // FIXME pass conifguration
```
Comment (JostCrow) [inline src/abstract-consumer.js]: Still a FIXME in the code

File: src/http-response-message.js [hunk 2]
```diff
@@ -0,0 +1,38 @@
+/**
+ * Response object modeled after aurelia-http-client HTTPResponseMessage.
+ * @class
+ */
+export class HttpResponseMessage {
```
Comment (JostCrow) [inline src/http-response-message.js]: Shouldn't this be AxiosHTTPResponseMessage?

File: src/abstract-consumer.js (hunk 1 above)
Comment (svenvandescheur) [reply src/abstract-consumer.js]: thx

File: src/http-response-message.js (hunk 2 above)
Comment (svenvandescheur) [reply src/http-response-message.js]: No, this mimics https://github.com/aurelia/http-client/blob/master/src/http-response-message.js for backwards compatibility.

File: src/abstract-consumer.js (hunk 1 above)
Comment (sergei-maertens) [reply src/abstract-consumer.js]: also docstring above needs updating

**Comment by svenvandescheur** [comment]
This also resolves #17.

**Pull request merged**